File: 05_Arrays/src/Check_if_the_Sentence_Is_Pangram_1832.js
/**
 * @param {string} sentence
 * @return {boolean}
 */
var checkIfPangram = function(sentence) {
    if (sentence.length < 26) {
        return false;
    }
    let count = 0;
    const obj = {};

    for (let i = 0; i < sentence.length; i++) {
        let letter = sentence[i];
        if (!obj[letter]) {
            obj[letter] = 1;
            count++;
        }
    }

    return count >= 26;
};

File: 05_Arrays/src/Check_if_the_Sentence_Is_Pangram_1832.java
import java.util.HashMap;

public class Check_if_the_Sentence_Is_Pangram_1832 {
    public static void main(String[] args) {
        System.out.println(checkIfPangram("thequickbrownfoxjumpsoverthelazydog"));
        System.out.println(checkIfPangram("xyz"));
    }
    public static boolean checkIfPangram(String sentence) {
        char[] arr = sentence.toCharArray();

        if (arr.length < 26) {
            return false;
        }

        int count = 0;
        HashMap<Character,Integer> charMap = new HashMap<>();

        for (char i : arr) {
            if (!charMap.containsKey(i)) {
                count++;
                charMap.put(i,1);
            }
        }

        return  count >= 26;
    }
}
